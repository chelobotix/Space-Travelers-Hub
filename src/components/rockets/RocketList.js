import { v4 as uuidv4 } from "uuid";
import { useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import Rocket from "./Rocket";
import {
  increment,
  fetchRockets,
} from "../../redux/features/rockets/rocketsSlice";

import style from "./RocketList.module.css";

const RocketList = () => {
  const dispatch = useDispatch();
  const state = useSelector((state) => state.rockets);

  useEffect(() => {
    dispatch(fetchRockets());
    console.log(state);
  }, [dispatch]);

  if (!state.rocketsArray) {
    return <h1>Loading...</h1>;
  }

  return (
    <div>
      <ul>
        {state.rocketsArray.map((rocket) => (
          <li key={uuidv4()}>
            <Rocket rocket={rocket} />
          </li>
        ))}
      </ul>
    </div>
  );
};

export default RocketList;
