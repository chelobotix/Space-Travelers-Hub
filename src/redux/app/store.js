import { configureStore } from '@reduxjs/toolkit';
import missionsReducer from '../features/missions/missionsSlice'
export const store = configureStore({
  reducer: {
    missions: missionsReducer,
  },
});
