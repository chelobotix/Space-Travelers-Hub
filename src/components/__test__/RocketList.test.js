import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import RocketList from '../rockets/RocketList';
import store from '../../redux/app/store';

describe('RocketsList', () => {
  test('RocketsList Tree', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <RocketList />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
