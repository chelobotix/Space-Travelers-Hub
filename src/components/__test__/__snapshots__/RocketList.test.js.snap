// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RocketsList RocketsList Tree 1`] = `
<h1>
  Loading...
</h1>
`;
